src/sentry/event_manager.py

```diff
@@ -888,6 +883,13 @@ def _save_aggregate(self, event, hashes, release, **kwargs):
         else:
             is_sample = can_sample
 
+        if not is_sample:
+            GroupHash.record_last_processed_event_id(
+                project.id,
+                [h.id for h in relevant_group_hashes],
+                event.event_id,
```

Comment:
I don't feel good about this either. This records that we've saved the event before we have actually saved it, but adding a fifth member to the return tuple of this method also feels wrong.